src/planeta/transactionHandler.js

```diff
@@ -0,0 +1,125 @@
+import { Tx, Util } from "leap-core";
+import JSBI from "jsbi";
+import { BigInt } from "jsbi-utils";
+import getConfig from "../config";
+
+const CONFIG = getConfig();
+const GOELLAR_COLOR = 3;
+const BITMASK_CO2 = BigInt("0xffffffff");
+const BITMASK_CO2_LOCKED = JSBI.leftShift(BITMASK_CO2, BigInt(32));
+const EarthContractData = require("./contracts/Earth.json");
+
+function parseHandshake(myAddress, inputs, outputs) {
+  const theirAddress = inputs.filter(
+    input => Util.isNST(input.color) && input.address !== myAddress
+  )[0].address;
+
+  // Filtering functions
+  const isOwner = owner => ({ address }) => owner === address;
+  const isMine = isOwner(myAddress);
+  const isTheirs = isOwner(theirAddress);
+  const isPassport = ({ color }) => Util.isNST(color);
+  const isGoellar = ({ color }) => color === GOELLAR_COLOR;
+
+  // Masking functions
+  const mask = bitmask => data => JSBI.bitwiseAnd(BigInt(data), bitmask);
+  const bitmaskCO2 = mask(BITMASK_CO2);
+  const bitmaskCO2Locked = mask(BITMASK_CO2_LOCKED);
+
+  // Defect detection
+  const getDefectInInputs = address => {
+    const defect = inputs.filter(isOwner(address)).filter(isGoellar);
+    return defect.length > 0 ? defect[0].value : BigInt(0);
+  };
+  const getDefectInPassport = (dataBefore, dataAfter) =>
+    JSBI.subtract(bitmaskCO2Locked(dataAfter), bitmaskCO2Locked(dataBefore));
+  const isDefect = (address, dataBefore, dataAfter) =>
+    JSBI.greaterThan(
+      JSBI.add(
+        getDefectInPassport(dataBefore, dataAfter),
+        getDefectInInputs(address, inputs)
+      ),
+      BigInt(0)
+    );
+
+  // Emissions and gains functions
+  const getCO2 = (dataBefore, dataAfter) =>
+    JSBI.toNumber(
+      JSBI.subtract(bitmaskCO2(dataAfter), bitmaskCO2(dataBefore))
+    ) / 1000;
+  const getGoellars = address =>
```

Comment:
What is going on in this function? 

- A comment would be nice
- Why do we have to do all of this math?
- What are the two magic constants `10000000000000000` and `100`?